### PR TITLE
feat: add go air project to pantry

### PR DESCRIPTION
**Pull request by mittalyashu**

### CODE REVIEW

File: projects/github.com/cosmtrek/air/package.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# * https://github.com/pkgxdev/pantry
+#
+# as a learning guide and reference. For example, if you are package a `go`
+# project, then browse the listings and find a go package.
+
+
+# this is the source tarball. Swap the version out for {{version}} or we won’t
+# be able to upgrade the package when it updates!
+distributable:
+  url: https://github.com/cosmtrek/air/archive/v{{version}}.tar.gz
```
Comment (jhheider) [inline projects/github.com/cosmtrek/air/package.yml:13]: should be:

https://github.com/cosmtrek/air/archive/refs/tags/v{{version}}.tar.gz

it looks like

File: projects/github.com/cosmtrek/air/package.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  # url: git+https://github.com/user/repo
+  # ref: v{{version}}}
+
+# if there’s a github then we can parse the versions
+versions:
+  github: cosmtrek/air
+
+platforms:
+  - darwin
+  - linux/x86-64
```
Comment (jhheider) [inline projects/github.com/cosmtrek/air/package.yml:32]: does it not build on linux+aarch64? please add a FIXME comment with the issue if so for future reference.

File: projects/github.com/cosmtrek/air/package.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+dependencies:
+  # to figure out dependencies you can read the project’s README
+  # however it is often easier to read the [brew formula](https://formulae.brew.sh/formula/)
+  #
+  # see dist.pkgx.dev for the packages pkgx provides
+  # if you need a package that isn’t there, you may need to build it!
+  #
+  # see semver.org for version specifications
+  go.dev: "*"
```
Comment (jhheider) [inline projects/github.com/cosmtrek/air/package.yml:42]: this is _probably_ just a build dependency, not a runtime dependency. but, all runtime dependencies are build dependencies, so if it's needed at runtime, no need to specify it again below.

**Comment by MarkusZoppelt** [comment]
@jhheider I couldn't push to @mittalyashu's branch, so I created #4116 with the relevant changes.

**Comment by mittalyashu** [comment]
Closing this PR for https://github.com/pkgxdev/pantry/pull/4116

**Pull request closed without merging**